src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2352,14 +2352,8 @@ private uint getClassGClayout(CORINFO_CLASS_STRUCT_* cls, byte* gcPtrs)
             uint result = 0;
 
             MetadataType type = (MetadataType)HandleToObject(cls);
-
-            int pointerSize = PointerSize;
-
-            int ptrsCount = AlignmentHelper.AlignUp(type.InstanceFieldSize.AsInt, pointerSize) / pointerSize;
-
-            // Assume no GC pointers at first
-            for (int i = 0; i < ptrsCount; i++)
-                gcPtrs[i] = (byte)CorInfoGCType.TYPE_GC_NONE;
+            uint size = type.IsValueType ? getClassSize(cls) : getHeapClassSize(cls);
+            new Span<byte>(gcPtrs, (int)((size + PointerSize - 1) / PointerSize)).Clear();
```

Comment:
This was copied from CoreCLR impl, it does a similar memset (see [this](https://github.com/dotnet/runtime/blob/c52177b21cb358d45a18a1f14d5855aedb61d0ec/src/coreclr/vm/jitinterface.cpp#L2422-L2425))

Previously, `getClassGClayout` used `InstanceFieldSize.AsInt` which for ref types always return just pointer-size while we need the real size.